### PR TITLE
Create ADR 007-signal-unsupportent-content-in-github.md

### PR DESCRIPTION
**Pull request by Rene2mt**
# Committer Notes

This PR adds an ADR, describing proposed approach for signaling legacy  unsupported content in the FedRAMP Automation repository.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- ~Have you added an explanation of what your changes do and why you'd like us to include them?~
- ~If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~
- ~If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+## Context
+
+The FedRAMP OSCAL automation team needs to cleanup the repository periodically, and establish a way to signal to the community areas of the repository that are no longer maintened (e.g., legacy proof-of-concept tools, outdated documentation / guidance, etc.).
+
+### Possible Solutions
+
+Signal unsupported code / content in the repository by:
+1. marking the content as **DEPRECATED** in a README
+2. **OBSOLETE** in a README
+ - deleting **OBSOLETE** content after some period of time or at specified release
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:18]: Thanks to the power of `git` and services like GitHub, GitLab, Codeberg, we get this for free.

```suggestion
 - deleting **OBSOLETE** content from future releases and leaving archival copies in an archive branch for posterity
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  * Update README clarifying intent of FedRAMP “template” and documenting known validation issues with each template.  Where possilbe, FedRAMP will attempt to fix template validation issues to reduce the amount of errors they produce.  
+* **documents** folder  
+  * Update README sending community to [https://automate.fedramp.gov](https://automate.fedramp.gov) for documentation going forward.  Add statement about PDF guides no longer being maintained and that existing GH issues submitted about the guides will be address via updates to [https://automate.fedramp.gov](https://automate.fedramp.gov)   
+    * **documents/rev4/README.md** \- mark as **OBSOLETE** and remove PDF guides
+    * **documents/rev5/README.md** \- mark as **OBSOLETE** and remove PDF guides
+* presentations folder  
+  * **presentations/data-bites**  
+    * Update with latest Data Bites slides  
+  * Add **presentations/oscal-implementers**
+    * Update with latest OSCAL Implementers slides
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:40]: Why not move to the automate.fedramp.gov site? These artifacts balloon git history and are really for documentation and communication, they are not code artifacts. Thoughts from the others?

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* **src/examples/README.md** \- **ARCHIVED**
+  * Move all current contents to archived subfolder (e.g., **src/examples/archived** )
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:42]: With git we can delete and still keep history, no?

```suggestion
  * Delete all current contents
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+* **src/utils/extensions-preview.xsl** \- **ARCHIVED**
+  * Move all current contents to archive subfolder (e.g., **src/examples/archived** )
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:44]: With git we can delete and still keep history, no?

```suggestion
  * Delete all current contents
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* **validations** folder \- **ARCHIVED**
+  * Move all current contents to archive subfolder (e.g., **src/validations/archived** )
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:46]: With git we can delete and still keep history, no?

```suggestion
  * Delete all current contents
```

**Review by aj-stein-gsa — COMMENTED** [review]
Some questions and initial thoughts on leaning into deletion. I would love feedback from you, Rene, and rest of team, and of course the community soon enough.

File: documents/adr/007-signal-unsupportent-content-in-github.md (hunk 2 above)
Comment (Rene2mt) [reply documents/adr/007-signal-unsupportent-content-in-github.md:40]: Good point.  I'll update the ADR accordingly

**Review by Rene2mt — COMMENTED** [review]
Will updated the PR with the requested changes shortly

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+Date: 2024-08-30
+
+## Status
+
+Pending Community Feedback
+
+## Context
+
+The FedRAMP OSCAL automation team needs to cleanup the repository periodically, and establish a way to signal to the community areas of the repository that are no longer maintened (e.g., legacy proof-of-concept tools, outdated documentation / guidance, etc.).
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:11]: Non-blocking: some more typos and cleanup.

```suggestion
The FedRAMP Automation Team must maintain organization of its repositories, including deprecation and migration information for data, tools, and documentation as part of its release strategy.
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+### Possible Solutions
+
+Signal unsupported code / content in the repository by:
+1. marking the content as **DEPRECATED** in a README
+2. **OBSOLETE** in a README
+ - deleting **OBSOLETE** content from future releases and leaving archival copies in an archive branch for posterity
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:18]:
```suggestion
1. Leverage [the current release strategy and semantic versioning](https://github.com/GSA/automate.fedramp.gov/blob/f085168bc485cf764a37c148d3a3109507d4695a/content/about/release.md):
  - identify deprecation of relevant data, tools, and documentation in README with an **OBSOLETE** label in the relevant READMEs;
    - make a minor release with deprecation information in those READMEs;
    - make a major release with deprecated content deleted and new data, tools, and documentation released
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+Pending Community Feedback
+
+## Context
+
+The FedRAMP OSCAL automation team needs to cleanup the repository periodically, and establish a way to signal to the community areas of the repository that are no longer maintened (e.g., legacy proof-of-concept tools, outdated documentation / guidance, etc.).
+
+### Possible Solutions
+
+Signal unsupported code / content in the repository by:
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:15]:
```suggestion
Several possible solutions for identifying deprecated and migrated data, tools, and documentation are below.

```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+1. marking the content as **DEPRECATED** in a README
+2. **OBSOLETE** in a README
+ - deleting **OBSOLETE** content from future releases and leaving archival copies in an archive branch for posterity
+3. marking the content as **ARCHIVED** and moving to an “archived” folder
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:19]:
```suggestion
1. Mark deprecated content by labelling relevant READMEs as **ARCHIVED** and move all relevant content to “archive” folder
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+Pending Community Feedback
+
+## Context
+
+The FedRAMP OSCAL automation team needs to cleanup the repository periodically, and establish a way to signal to the community areas of the repository that are no longer maintened (e.g., legacy proof-of-concept tools, outdated documentation / guidance, etc.).
+
+### Possible Solutions
+
+Signal unsupported code / content in the repository by:
+1. marking the content as **DEPRECATED** in a README
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:16]:
```suggestion
1. Do nothing, leave all data, tools, and documentation as-is
1. Marking the content as **DEPRECATED** in the relevant READMEs
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+2. **OBSOLETE** in a README
+ - deleting **OBSOLETE** content from future releases and leaving archival copies in an archive branch for posterity
+3. marking the content as **ARCHIVED** and moving to an “archived” folder
+   
+## Consequences
+
+The above will be applied to the current repository as follows:
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:23]:
```suggestion
For the purposes of this ADR, there is an inventory for this repository of different tools, data, documentation, and their future deprecation below.

```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
Apologies for the delayed feedback on this ADR. The exploration of solution space and and the inventory are good. I have some more editorial changes and think we ought to more closely relate it to the [current release strategy and semantic versioning](https://github.com/GSA/automate.fedramp.gov/blob/f085168bc485cf764a37c148d3a3109507d4695a/content/about/release.md), since this allows us to lean into the strengths of `git` and delete it in the future and let everyone use the older releases as archival references for free (including with durable links through git tags).

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+For the purposes of this ADR, there is an inventory for this repository of different tools, data, documentation, and their future deprecation below.
+
+* **dist/content/rev4**  
+  * Add README \- FR has transitioned to Rev 5\.  Rev 4 baselines, resources, and templates are no longer maintained but will be kept frozen for historical purposes.
+  * Mark as **DEPRECATED**
+* **dist/content/rev5/resources**  
+  * Archive README \- letting community know FedRAMP is transitioning from *fedramp\_extensions.xml* & *fedramp\_values.json*
+    * FedRAMP to consider future work on replacing them with report of constraints auto-generated by CI or OSCAL-CLI (from Metaschema constraints)  
+* **dist/content/rev5/templates**  
+  * Update README clarifying intent of FedRAMP “template” and documenting known validation issues with each template.  Where possilbe, FedRAMP will attempt to fix template validation issues to reduce the amount of errors they produce.
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:36]: Non-blocking question: what is the intent?

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    * FedRAMP to consider future work on replacing them with report of constraints auto-generated by CI or OSCAL-CLI (from Metaschema constraints)  
+* **dist/content/rev5/templates**  
+  * Update README clarifying intent of FedRAMP “template” and documenting known validation issues with each template.  Where possilbe, FedRAMP will attempt to fix template validation issues to reduce the amount of errors they produce.  
+* **documents** folder  
+  * Update README sending community to [https://automate.fedramp.gov](https://automate.fedramp.gov) for documentation going forward.  Add statement about PDF guides no longer being maintained and that existing GH issues submitted about the guides will be address via updates to [https://automate.fedramp.gov](https://automate.fedramp.gov)   
+    * **documents/rev4/README.md** \- mark as **OBSOLETE** and remove PDF guides
+    * **documents/rev5/README.md** \- mark as **OBSOLETE** and remove PDF guides
+* presentations folder  
+  * **presentations/data-bites**  
+    * Move to automte.fedramp.gov and update with latest Data Bites slides
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:43]: Minor, non-blocking: the domain is misspelled.

```suggestion
    * Move to [https://automate.fedramp.gov](https://automate.fedramp.gov) and update with latest Data Bites slides  
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

File: documents/adr/007-signal-unsupportent-content-in-github.md (hunk 12 above)
Comment (Rene2mt) [reply documents/adr/007-signal-unsupportent-content-in-github.md:36]: Community often has questions about errors when trying to validate the templates.   I've been able to find and fix a lot of the SSP template validation failures using our new tooling, but there's a handful of errors that remain (some seem like false positives).  I'm thinking we should document those remaining errors somewhere, so the community knows these are known issues.

Comment (aj-stein-gsa) [reply documents/adr/007-signal-unsupportent-content-in-github.md:36]: Per conversation with you and Dave on Friday, we can put a deprecation warning and defer deletion as necessary, it doesn't necessarily have to be immediately. I will close this out for now.

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
+# 7. Signal Unsupported Contents in the Repository
+
+Date: 2024-08-30
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:3]: Important to change the date accordingly.

```suggestion
Date: 2024-09-17
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+## Status
+
+Pending Community Feedback
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:7]: Ditto, mark it accepted as no feedback has been received by community over two meetings.

```suggestion
Accepted
```

File: documents/adr/007-signal-unsupportent-content-in-github.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+Several possible solutions for identifying deprecated and migrated data, tools, and documentation are below.
+
+1. Do nothing, leave all data, tools, and documentation as-is
+1. Marking the content as **DEPRECATED** in the relevant READMEs
+1. Leverage [the current release strategy and semantic versioning](https://github.com/GSA/automate.fedramp.gov/blob/f085168bc485cf764a37c148d3a3109507d4695a/content/about/release.md):
+  - identify deprecation of relevant data, tools, and documentation in README with an **OBSOLETE** label in the relevant READMEs;
+    - make a minor release with deprecation information in those READMEs;
+    - make a major release with deprecated content deleted and new data, tools, and documentation released
+1. Mark deprecated content by labelling relevant READMEs as **ARCHIVED** and move all relevant content to “archive” folder
+
```
Comment (aj-stein-gsa) [inline documents/adr/007-signal-unsupportent-content-in-github.md:24]: It is time to make a decision, so we shall picture Solution 3.

```suggestion
 ## Decision
 
 The FedRAMP Automation Team will implement Solution 3: identify tools, data, and content for migration, deprecation, remove content accordingly by saving them in an archival minor release and deleting them in a major version release.
   
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
Minor review feedback, getting ready for merge.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**